keyboards/hardwareabstraction/handwire/keymaps/default/keymap.c

```diff
@@ -0,0 +1,334 @@
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,     KC_TRNS,    KC_TRNS,      KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                                 KC_TRNS,                           KC_TRNS,     KC_TRNS,     KC_HPTDWLD,     KC_TRNS,     KC_HPTDWLI
+    ),
+    [3] = LAYOUT_60_arrows(
+        KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,  KC_TRNS, KC_TRNS,  KC_TRNS,
+        KC_TRNS,  KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,  KC_TRNS, KC_TRNS,  KC_TRNS,
+        KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,          KC_TRNS,
+        KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,   KC_TRNS,     KC_TRNS,    KC_TRNS,      KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS,                                 KC_TRNS,                           KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS,     KC_TRNS
+    )
+};
+
+// Documentation: custom_quantum_functions.md
+void keyboard_post_init_user(void){
+    #ifdef BUZZER_ENABLE
+    setPinOutput(BUZZER_PIN);
+    #endif
+}
+
+void housekeeping_task_user(void){
+    #ifdef BUZZER_ENABLE
+    if(buzzer_on){
+        if(buzzer_active && timer_elapsed(buzzer_timer) > buzzer_dwell){
+        buzzer_active = false;
+        writePinLow(BUZZER_PIN);
+        }
+    }
+    #endif
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed){                         // true when pressed, false when released
+        initial_keypress = true;
+
+        #ifdef BUZZER_ENABLE
+        if(buzzer_on){
+            if(!buzzer_active){
+                buzzer_active = true;
+                buzzer_timer = timer_read();
+                writePinHigh(BUZZER_PIN);
+            }
+        }
+        #endif
+
+        switch (keycode) {
+            #ifdef HAPTIC_ENABLE
+            case KC_HPTON:
+                haptic_enable();
+                break;
+
+            case KC_HPTOFF:
+                haptic_disable();
+                break;
+
+            case KC_HPTTOG:
+                haptic_toggle();
+                break;
+
+            case KC_HPTRST:
+                haptic_reset();
+                break;
+
+            case KC_HPTFBK:
+                haptic_feedback_toggle();
+                break;
+
+            case KC_HPTCONT:
+                haptic_toggle_continuous();
+                break;
+
+            case KC_HPTDWLI:
+                haptic_dwell_increase();
+                break;
+
+            case KC_HPTDWLD:
+                haptic_dwell_decrease();
+                break;
+            #endif
+
+            #ifdef BUZZER_ENABLE
+            case KC_BZTOG:
```

Comment:
Is there a reason for this extra code?  It looks like it's a lot of duplication of the solenoid code. 

And if that is the case, the solenoid code actually supports multiple pins. `SOLENOID_PINS { E6, B2 }` could be used.

Reply:
The reason I didn't declare another solenoid pin was because I want to be able to individually turn off the buzzer and the solenoid. From my knowledge, it didn't look like there was a way to turn off an individual pin while keeping the other one on.